src/include/migraphx/algorithm.hpp

```diff
@@ -95,20 +95,24 @@ inline size_t levenshtein_distance(const std::string& s1, const std::string& s2)
     const size_t l1 = s1.length();
     const size_t l2 = s2.length();
 
-    std::vector<std::vector<size_t>> d(l1 + 1, std::vector<size_t>(l2 + 1));
+    std::vector<size_t> d0(l2 + 1), d1(l2 + 1);
 
-    for(size_t i = 1; i <= l1; i++)
-        d[i][0] = i;
     for(size_t j = 1; j <= l2; j++)
-        d[0][j] = j;
+        d0[j] = j;
```

Comment:
Use `std::iota`.

Reply:
We are explicitly using the indices here inside the loop. Those index values are very important and therefore doing a duplicate index calculation in iota controlled loops isn't most efficient here.. -- unless I have missed something. We need `i, i-1, j, j-1.`  in these two (nested) for loops. Thanks.

Reply:
You are setting the vector to `0,1,2,3,4,...` in this loop, so doing `std::iota(d1.begin(), d1.end(), 0)` is equivalent. We [prefer standard algorithms over raw loops](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/wiki/Developer's-Guide#raw-loops) when its available.

Reply:
(Totally agreed with your sentiment.., but.. )

I wouldn't (logically speaking) set an element of `vector<size_t>` to a `value type`: `vector<size_t>::iterator`, line # 106:
`d1[0] = i;`
The index `i` is also the cost of levenshtein_distance (boundary condition).
Thanks.